synthesizer/process/src/cost.rs

```diff
@@ -395,6 +419,49 @@ pub fn cost_in_microcredits<N: Network>(stack: &Stack<N>, function_name: &Identi
         })
 }
 
+/// Returns the minimum number of microcredits required to run the finalize (depcrated).
+pub fn cost_in_microcredits_deprecated<N: Network>(stack: &Stack<N>, function_name: &Identifier<N>) -> Result<u64> {
+    // Retrieve the finalize logic.
+    let Some(finalize) = stack.get_function_ref(function_name)?.finalize_logic() else {
+        // Return a finalize cost of 0, if the function does not have a finalize scope.
+        return Ok(0);
+    };
+    // Get the cost of finalizing all futures.
+    let mut future_cost = 0u64;
+    for input in finalize.inputs() {
+        if let FinalizeType::Future(future) = input.finalize_type() {
+            // Get the external stack for the future.
+            let stack = stack.get_external_stack(future.program_id())?;
+            // Accumulate the finalize cost of the future.
+            future_cost = future_cost
+                .checked_add(cost_in_microcredits_deprecated(stack, future.resource())?)
+                .ok_or(anyhow!("Finalize cost overflowed"))?;
+        }
+    }
+    // Aggregate the cost of all commands in the program.
+    finalize
+        .commands()
+        .iter()
+        .map(|command| cost_per_command(stack, finalize, command))
+        .try_fold(future_cost, |acc, res| {
+            res.and_then(|x| acc.checked_add(x).ok_or(anyhow!("Finalize cost overflowed")))
+        })
+        .and_then(|v2_cost| {
+            let mut additional_mapping_cost = 0u64;
+            for command in finalize.commands() {
+                match command {
+                    Command::Get(_) | Command::GetOrUse(_) | Command::Contains(_) => {
+                        additional_mapping_cost = additional_mapping_cost
+                            .checked_add(MAPPING_BASE_COST_V1 - MAPPING_BASE_COST_V2)
+                            .ok_or(anyhow!("Mapping cost overflowed"))?;
+                    }
+                    _ => {}
```

Comment:
Instead of this brute force approach of subtracting, we can change the function signature of `cost_per_command` to accept a `const generic` or another input to determine which consensus version to use (V1 or V2). And then inside `cost_per_command` you can do the following:

```
let mapping_base_cost = match consensus_version {
    v1 => MAPPING_BASE_COST_V1
    v2 => MAPPING_BASE_COST_V2
};
```

Reply:
I had initially thought this would be fine since the intension was to remove the function in the future, but I believe we should go with the a cleaner and safer approach regardless. 

This change should be fairly quick.